azurerm/internal/services/network/data_source_virtual_machine_scale_set_network_interfaces.go

```diff
@@ -0,0 +1,275 @@
+)
+
+func dataSourceArmVirtualMachineScaleSetNetworkInterfaces() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmVirtualMachineScaleSetNetworkInterfacesRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"virtual_machine_scale_set_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"network_interfaces": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"network_security_group_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"mac_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"virtual_machine_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"ip_configuration": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"subnet_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"private_ip_address": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"private_ip_address_version": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"private_ip_address_allocation": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"public_ip_address_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"application_gateway_backend_address_pools_ids": {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+
+									"load_balancer_backend_address_pools_ids": {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+
+									"load_balancer_inbound_nat_rules_ids": {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+
+									"application_security_group_ids": {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
```

Comment:
Since this is a Computed attribute, we should just use TypeList instead of TypeSet. And this would also benefit the users when they are trying to use one of the element.